source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,166 @@ namespace aspect
 
     return (material_model->get_model_dependence().viscosity != MaterialModel::NonlinearDependence::none);
   }
-}
 
+
+
+  template <int dim>
+  void Simulator<dim>::apply_limiter_to_dg_solutions (const AdvectionField &advection_field)
+  {
+    AssertThrow (dim == 2,
+                 ExcMessage ("Only apply bound preserving limiter to the 2 dimensional problem"));
```

Comment:
I was planning to use tensor product to denote those weights and points, and then call the formula like below
Quadrature< dim >::Quadrature	( const std::vector< Point< dim > > & points,
const std::vector< double > & 	weights )	

Since I never use this function before, and also I need to test it with a 3D example, I just decided to implement the 3D case after this pull.

Do you think I am on the right way to go?